sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -193,7 +203,104 @@ public Mono<CosmosItemResponse<Object>> deleteItem(String itemId,
                                                        PartitionKey partitionKey,
                                                        CosmosItemRequestOptions requestOptions) {
 
-        return container.deleteItem(itemId, partitionKey, requestOptions);
+        return deleteItemInternal(itemId, partitionKey, requestOptions);
+    }
+
+    private Mono<CosmosItemResponse<Object>> deleteItemInternal(String itemId, PartitionKey partitionKey, CosmosItemRequestOptions requestOptions) {
+        this.encryptionProcessor.initEncryptionSettingsIfNotInitializedAsync();
+        return Mono.just(this.encryptionProcessor.getEncryptionSettings())
+            .flatMap(settings -> {
+                try {
+                    return Mono.zip(
+                        checkAndGetEncryptedId(itemId, settings),
+                        checkAndGetEncryptedPartitionKey(partitionKey, settings)
+                    ).flatMap(encryptedIdPartitionTuple -> container.deleteItem(encryptedIdPartitionTuple.getT1(), encryptedIdPartitionTuple.getT2(), requestOptions));
+                } catch (Exception ex) {
+                    return Mono.error(ex);
+                }
+            });
+    }
+
+    private Mono<String> checkAndGetEncryptedId(String itemId, EncryptionSettings encryptionSettings)
+    {
+        if (this.encryptionProcessor.getClientEncryptionPolicy().getIncludedPaths().stream().
+            anyMatch(includedPath -> includedPath.getPath().substring(1).equals(Constants.PROPERTY_NAME_ID))) {
+            return this.getEncryptedItem(encryptionSettings, Constants.PROPERTY_NAME_ID, itemId);
+        }
+        return Mono.just(itemId);
+    }
+
+    private Mono<PartitionKey> checkAndGetEncryptedPartitionKey(PartitionKey partitionKey, EncryptionSettings encryptionSettings) {
```

Comment:
Please add dedicated unit tests for this method going through PartitonKey NULL/EMPTY/null as well as single and hierarchical partition keys with dedicated encryption and because parent node is part of the encryption policy.

Reply:
I have partition key test for null/empty and single pk. I don't have test cases for the hierarchical pks. I will add those and update this comment.

Reply:
@aayush3011 Is this done?

Reply:
Yes